shardingsphere-features/shardingsphere-shadow/shardingsphere-shadow-core/src/main/java/org/apache/shardingsphere/shadow/rewrite/parameter/ShadowParameterRewriter.java

```diff
@@ -41,4 +41,8 @@ public final boolean isNeedRewrite(final SQLStatementContext sqlStatementContext
     }
     
     protected abstract boolean isNeedRewriteForShadow(SQLStatementContext sqlStatementContext);
+    
+    protected String getShadowColumn() {
+        return shadowRule.getColumn();
```

Comment:
Why do we need another function?

Reply:
Each subclass is called multiple times, and protected methods are extracted from the parent class.